config-store-yaml/src/main/java/io/stargate/config/store/api/yaml/ConfigStoreYaml.java

```diff
@@ -18,22 +18,36 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.type.MapType;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Ticker;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
```

Comment:
Does this come from guava? If so that'll cause issues due the guava incompatibilities we have in different modules.

Reply:
Should private libs be isolated from each other by OSGi?

Reply:
Not if everything imports config-store-yaml

Reply:
Yes, it uses guava but other modules are using it as well.
We have an influence over the guava version that is used by all extensions. If we will keep the version consistent there should be no problems, isn't it? (currently, the newest version that we are using is: `29.0-jre`)

Reply:
Shouldn't dependent modules import `config-store-api`? I'd expect `config-store-yaml` to be a leaf module providing config API implementation... why would ordinary modules depend specifically on the "yaml" config impl?

Reply:
yes, agree - Importing api should be enough. I will soon create a PR with the usage of the config-store from the Kafka-producer.